### ISSUE TITLE
`v : ()->nil` parsed as feature declaration

### ISSUE THREAD

**Issue by maxteufel** [tokiwa-software/fuzion#1149]
```
ex is
  match_issue =>
    o option i32 := 42

    match o
      val i32 => val : ()->nil
      nil => 1 : ()->match_issue
```

the errors are:

```
ex.fz:6:24: error 1: Syntax error: expected identifier name, infix/prefix/postfix operator, 'ternary ? :', 'index' or 'set' name, found left parenthesis '('
      val i32 => val : ()->nil
-----------------------^
While parsing: name, parse stack: name (2 times), call, callList, inherit, inherits, routOrField, feature, stmnt, stmnts, block, caseBlock, caze, cases, match, term, opTail, opExpr, expr, stmnt, stmnts, block, implRout, implFldOrRout, routOrField, feature, stmnt, stmnts, block, implRout, implFldOrRout, routOrField, feature, stmnt, stmnts, unit


ex.fz:6:31: error 2: No line break may occur at this position
      val i32 => val : ()->nil
------------------------------^
This code is part of an expression that must reside within a single line.
While parsing: implRout, parse stack: implRout, routOrField, feature, stmnt, stmnts, block, caseBlock, caze, cases, match, term, opTail, opExpr, expr, stmnt, stmnts, block, implRout, implFldOrRout, routOrField, feature, stmnt, stmnts, block, implRout, implFldOrRout, routOrField, feature, stmnt, stmnts, unit
To solve this, enclose the expression in parentheses '(' and ')'.


ex.fz:6:18: error 3: Duplicate feature declaration
      val i32 => val : ()->nil
-----------------^
Feature that was declared repeatedly: 'ex.match_issue.val'
originally declared at ex.fz:6:7:
      val i32 => val : ()->nil
------^
To solve this, consider renaming one of these two features, e.g., as 'valʼ' (using a unicode modifier letter apostrophe 'ʼ' U+02BC) or adding an additional argument (e.g. '_ unit' for an ignored unit argument used only to disambiguate these two).


ex.fz:5:5: error 4: Incompatible types in cases of match statement
    match o
----^
Incompatible result types in different branches:
block returns value of type 'unit' at ex.fz:6:18:
      val i32 => val : ()->nil
-----------------^
block returns value of type 'list i32' at ex.fz:7:16:
      nil => 1 : ()->match_issue
---------------^


ex.fz:6:26: error 5: No type information can be inferred from a lambda expression
      val i32 => val : ()->nil
-------------------------^
A lambda expression can only be used if assigned to a field or argument of type 'Function'
with argument count of the lambda expression equal to the number of type parameters of the type.  The type of the
assigned field must be given explicitly.
To solve this, declare an explicit type for the target field, e.g., 'f (i32, i32) -> bool := x, y -> x > y'.

5 errors.

```

a workaround is to put parentheses around `val : ()->nil`.

**Comment by fridis** [comment]
The problem is that the parser tries to parse `val : ()->nil` as a feature declaration for `val` that inherits from `()->nil`. A workaround is to wrikte `(val : ()->nil)`. 
For a proper fix, I see three possibilities
- delay the decision if we parse feature declaration or an expression until we find `is`, `=>` or `:=`
- use a keyword `inherits` instead of `:`
- do not allow use `:` as an infix operator

**Comment by fridis** [comment]
Here is a smaller example
```
ex is
  v := 42
  m => v : ()->nil
```

**Comment by fridis** [comment]
This is fixed with #1226.
